### DISCUSSION TITLE
middleware + strategy: "database" impossible (NOT on edge)?

### DISCUSSION THREAD

**Discussion by subproject22** [nextauthjs/next-auth#11108] [Help]
Prisma adapter (SQLite)

Running locally in dev. Never planned to host on edge. Don't have `runtime = "edge"`

When I add middleware like

```
import { auth } from "@/auth";
import { NextRequest, NextResponse } from "next/server";

export default auth((req) => {
  console.log("middleware", req.auth);
  return NextResponse.next();
});

export const config = {
  matcher: "/posts",
};
```

Then when I hit protected route I got error in console

```
[auth][error] AdapterError: Read more at https://errors.authjs.dev#adaptererror
[auth][cause]: Error: PrismaClient is not configured to run in Vercel Edge Functions or Edge Middleware. In order to run Prisma Client on edge runtime, either:

[auth][error] SessionTokenError: Read more at https://errors.authjs.dev#sessiontokenerror
[auth][cause]: Error: PrismaClient is not configured to run in Vercel Edge Functions or Edge Middleware. In order to run Prisma Client on edge runtime, either:
```

If I change to `strategy: "jwt"` or disable middleware, it works.

I've read https://authjs.dev/guides/edge-compatibility

But this doesn't explain why it throws error when I DON'T host on edge and not going to?

Basically middleware will not work with `strategy: "database"` on ANY environment (not only edge)? Otherwise, why it shows me error locally and how to fix it?
